acme/acme.go

```diff
@@ -100,6 +100,10 @@ type Client struct {
 	// will have no effect.
 	DirectoryURL string
 
+	// ShouldRetry reports whether a request can be retried based on the HTTP response status code.
+	// When ShouldRetry is nil, the default behavior will take precedence.
+	ShouldRetry func(httpStatusCode int) bool
```

Comment:
It would be nice to take the opportunity to make this more generic, as in make this take a `http.Response` object. Would be more configurable.

Reply:
Btw, where is the config for number of retries?

Reply:
in http

Reply:
yep it can take `*http.Response`

Reply:
Nice! 👍